Develop3

alenaa-bva opened the pull request.

alenaa-bva committed:
Добавлены файлы POM и тестов.
Добавлены тесты для home_page

alenaa-bva committed:
Добавлены тесты для страницы восстановления пароля

добавлены тесты для ЛК

alenaa-bva committed:
Закончены тесты
Прикреплен отчет

KseJha commented on pages/base_page.py:
```diff
@@ -0,0 +1,117 @@
+    def __init__(self, driver):
+        self.driver = driver
+
+    def click_on_the_element(self, driver, xpath):
```
Нужно исправить тут и далее во всех page object классах: передавать драйвер в метод не нужно, драйвер уже есть в базовом классе и вызывается через self. То есть драйвер передается только один раз при иницилизации класса

KseJha commented on tests/conftest.py:
```diff
@@ -0,0 +1,27 @@
+    if request.param == 'chrome':
+        webdriver = chrome
+        webdriver.maximize_window()
+        firefox.quit()
```
Нужно исправить: не надо делать такую фикстуру, в которой открываются два браузера и один потом надо закрывать, так как это занимает много времени и ресурсов (важно на больших проектах). Лучше убрать отдельные фикстуры для хрома и файерфокса, в одной фикстуре driver делать инициализацию нужного драйвера по условию (условие уже написано)

KseJha commented on xpath_data.py:
```diff
@@ -0,0 +1,68 @@
+class BasePagePaths:
```
Нужно исправить: это файл должен называться locators

KseJha commented on data.py:
```diff
@@ -0,0 +1,14 @@
+    LOGIN_PAGE_URL = f"{BASE_PAGE_URL}login"
+    FEED_PAGE_URL = f"{BASE_PAGE_URL}feed"
+    REGISTER_PAGE_URL = f"{BASE_PAGE_URL}register"
+    FORGOT_PASSWORD_PAGE_URL = f"{BASE_PAGE_URL}forgot-password"
```
Можно лучше: урлы не относятся к данным, а скорее к конфигурации, поэтому лучше их вынечти в отдельный модуль

KseJha commented on pages/base_page.py:
```diff
@@ -0,0 +1,117 @@
+                    """
+            self.driver.execute_script(script, from_element, to_element)
+
+    def login(self, driver):
+
+        register_data = generate_register_data()
+
+        driver.get(BasePageData.REGISTER_PAGE_URL)
+
+        self.fill_the_field(driver, RegistrationPagePaths.rp_name_input, register_data["name"])
+        self.fill_the_field(driver, RegistrationPagePaths.rp_email_input, register_data["login"])
+        self.fill_the_field(driver, RegistrationPagePaths.rp_password_input, register_data["password"])
+        self.click_on_the_element(driver, RegistrationPagePaths.sign_up_button)
+
+        driver.get(BasePageData.LOGIN_PAGE_URL)
+
+        self.fill_the_field(driver, LoginPagePaths.lp_email_input, register_data["login"])
+        self.fill_the_field(driver, LoginPagePaths.lp_password_input, register_data["password"])
+        self.click_on_the_element(driver, LoginPagePaths.login_button)
+
+        self.wait_element_to_be_visible(driver, HomePagePaths.collect_a_burger_header)
```
Нужно исправить: этот метод должен лежать в какм-то другом page object классе, например, login_page.
К тому же в этом методе можно выделить два метода - регистрация и логин, так что их тоже стоит разделить

alenaa-bva committed:
Убрала передачу драйвера в wait_element_to_be_visible.

Переименовала файл с локаторами в locators.py

Логин и регистрацию перенесла в отдельный файл login_page.py

alenaa-bva committed:
Убрала передачу драйвера в методах

alenaa-bva committed:
класс BasePageData переименован в UrlLib

alenaa-bva committed:
удалены ненужные вызовы self.driver

alenaa-bva committed:
итоговый коммит с работающими тестами

alenaa-bva committed:
Удалена папка .idea из корня проекта
